### PR TITLE
Adding support for serde

### PR DESCRIPTION
**Pull request by michaelciraci**
This adds an optional serde feature.

### CODE REVIEW

**Review by Lokathor — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -19,9 +19,11 @@ default = ["std"]
 # Activate `std` within the crate. Currently this gives a much faster `sqrt`
 # impl when an explicit hardware sqrt isn't available.
 std = []
+serde = ["dep:serde"]
```
Comment (Lokathor) [inline Cargo.toml:22]: oh we need to have this feature removed and then have the `serde` feature just be the optional dep only, at least based on our current MSRV we do.

Comment (waywardmonkeys) [reply Cargo.toml:22]: Or rebase forward since this feature was added in Rust 1.60 and that's the new MSRV :)

File: Cargo.toml [hunk 2]
```diff
@@ -19,9 +19,11 @@ default = ["std"]
...
 
 [dependencies]
 safe_arch = { version = "0.7", features = ["bytemuck"] }
+serde = { version = "1", optional = true }
```
Comment (waywardmonkeys) [inline Cargo.toml:26]: Should be safe to do a `default-features = false` in here as well.

File: Cargo.toml (hunk 1 above)
Comment (michaelciraci) [reply Cargo.toml:23]: I just rebased so hopefully we should now be good

**Comment by michaelciraci** [comment]
Rebased with new CI changes

**Comment by Lokathor** [comment]
released `0.7.27`

**Pull request merged**